.github/workflows/netsim_runner.yaml

```
@@ -0,0 +1,254 @@
+  MSRV: "1.66"
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
+  IROH_FORCE_STAGING_RELAYS: "1"
+
+jobs:
+  netsim:
+    name: Netsim
+    timeout-minutes: 45
+    runs-on: [self-hosted, linux, X64]
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+        ref: ${{ inputs.branch }}
+
+    - name: Install rust stable
+      uses: dtolnay/rust-toolchain@stable
+
+    - name: Install sccache
+      uses: mozilla-actions/sccache-action@v0.0.5
+
+    - name: Build iroh
+      run: |
+        cargo build --profile ${{ inputs.build_profile }} --workspace --all-features
+
+    - name: Fetch and build chuck
+      run: |
+        cd ..
+        rm -rf chuck
+        git clone --single-branch --branch ${{ inputs.netsim_branch }} https://github.com/n0-computer/chuck.git
+        cd chuck
+        cargo build --release
+
+    - name: Install netsim deps
+      run: |
+        cd ../chuck/netsim
+        sudo apt update
+        ./setup.sh
+        ./cleanup.sh || true
+
+    - name: Copy binaries to right location
+      run: |
+        cp target/${{inputs.build_profile}}/iroh ../chuck/netsim/bins/iroh
+        cp target/${{inputs.build_profile}}/iroh-relay ../chuck/netsim/bins/iroh-relay
+        cp ../chuck/target/release/chuck ../chuck/netsim/bins/chuck
+
+    - name: Get commit sha
+      shell: bash
+      run: |
+        echo "LAST_COMMIT_SHA=$(git rev-parse --short ${GITHUB_SHA})" >> ${GITHUB_ENV}
+  
+    - name: Run tests
+      run: |
+        cd ../chuck/netsim
+        # split sim_paths by comma
+        IFS=',' read -ra sim_paths <<< "${{ inputs.sim_paths }}"
+        for sim_path in "${sim_paths[@]}"; do
+          sudo python3 main.py ${{ inputs.debug_logs && '--debug' || ''}} ${{ inputs.visualizations && '--visualize' || ''}} --max-workers=${{ inputs.max_workers }} --integration $sim_path
+        done
+
+    - name: Generate report
+      id: generate_report
+      run: |
+        cd ../chuck/netsim
+        python3 reports_csv.py --table > report_table.txt
+        python3 reports_csv.py --prom --commit ${{ env.LAST_COMMIT_SHA }} > report_prom.txt
+        python3 reports_csv.py --metro --commit ${{ env.LAST_COMMIT_SHA }} > report_metro.txt
+        python3 reports_csv.py --metro --integration --commit ${{ env.LAST_COMMIT_SHA }} > report_metro_integration.txt
+    
+    - name: Dump report
+      run: |
+        export AWS_ACCESS_KEY_ID=${{secrets.S3_ACCESS_KEY_ID}}
+        export AWS_SECRET_ACCESS_KEY=${{secrets.S3_ACCESS_KEY}}
+        export AWS_DEFAULT_REGION=us-west-2
+        
+        curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+        unzip -q awscliv2.zip
+        sudo ./aws/install --update
+
+        cd ../chuck/netsim
+
+        aws_fname=${{ env.LAST_COMMIT_SHA }}.tar.gz
+        tar cvzf report.tar.gz report_prom.txt report_table.txt report_metro.txt report_metro_integration.txt logs/ report/ viz/
```

```suggestion
        tar -cvzf report.tar.gz report_prom.txt report_table.txt report_metro.txt report_metro_integration.txt logs/ report/ viz/
```
doesn't really matter, but we have modern tools these days.  no need to pretend they're stuck in the 80s ;)